akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/internal/TestProbeImpl.scala

```diff
@@ -224,6 +228,10 @@ private[akka] final class TestProbeImpl[M](name: String, system: ActorSystem[_])
     expectMessageClass_internal(max.asScala.dilated, clazz)
 
   private def expectMessageClass_internal[C](max: FiniteDuration, c: Class[C]): C = {
+    if (classOf[Signal].isAssignableFrom(c)) {
+      throw new IllegalArgumentException(
+        s"${c.getName} is a signal, expecting signals with a TestProbe is not possible")
+    }
```

Comment:
Fail fast if you try to expect a signal, which will never work, related to #26535 (but that talks about the sending side)